### PR TITLE
Upgrade Node/NPM

### PR DESCRIPTION
**Pull request by chad-ramsey-ck**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/thrift-client-timing-filter/tsconfig.json [hunk 1]
```diff
@@ -12,7 +12,8 @@
         "strict": true,
         "noUnusedLocals": true,
         "pretty": true,
-        "removeComments": true
+        "removeComments": true,
+        "useUnknownInCatchVariables": false
```
Comment (chad-ramsey-ck) [inline packages/thrift-client-timing-filter/tsconfig.json:16]: Without `"useUnknownInCatchVariables": false` tsc continued to revert `err` types to become typeless despite being updated to `any`. Assuming this has to do with `"strict": true`. An alternative to this appears to be casting `err` types to `Error` before accessing them (i.e., `((e as Error).message)`).

File: packages/thrift-integration/src/client/connections/HttpConnection.spec.ts [hunk 2]
```diff
@@ -601,7 +601,7 @@ describe('HttpConnection', () => {
         })
 
         after(async () => {
-            return new Promise((resolve, reject) => {
+            return new Promise<void>((resolve, reject) => {
```
Comment (chad-ramsey-ck) [inline packages/thrift-integration/src/client/connections/HttpConnection.spec.ts:604]: Addressing breaking change: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-1.html#resolves-parameters-are-no-longer-optional-in-promises

File: packages/thrift-server-core/src/main/binary.ts [hunk 3]
```diff
@@ -27,7 +24,7 @@ export function readByte(byte: number): number {
  * @param offset Offset
  */
 export function readI16(bytes: Buffer, offset: number = 0): number {
-    return bytes.readInt16BE(offset, noAssert)
```
Comment (chad-ramsey-ck) [inline packages/thrift-server-core/src/main/binary.ts:30]: Buffer read/write functions no longer accept `noAssert` as an argument from > Node 10: https://nodejs.org/docs/latest-v10.x/api/buffer.html#buffer_buf_readint16le_offset.

File: packages/thrift-server-core/src/main/protocols/BinaryProtocol.ts [hunk 4]
```diff
@@ -167,7 +167,7 @@ export class BinaryProtocol extends TProtocol {
 
     public writeStringOrBinary(
         name: string,
-        encoding: string,
+        encoding: BufferEncoding,
```
Comment (chad-ramsey-ck) [inline packages/thrift-server-core/src/main/protocols/BinaryProtocol.ts:170]: `encoding` must now use the string literal type.

File: packages/thrift-server-core/src/main/protocols/CompactProtocol.ts [hunk 5]
```diff
@@ -621,12 +621,12 @@ export class CompactProtocol extends TProtocol {
      * Convert from zigzag long to long.
      */
     private zigzagToI64(i64: Int64): Int64 {
-        let hi = i64.buffer.readUInt32BE(0, true)
-        let lo = i64.buffer.readUInt32BE(4, true)
+        let hi = i64.buffer.readUInt32BE(0)
```
Comment (chad-ramsey-ck) [inline packages/thrift-server-core/src/main/protocols/CompactProtocol.ts:624]: See above. `noAssert` no longer supported.

**Review by David-Hoyt-CK — APPROVED** [review]

**Review by danielchencreditkarma — APPROVED** [review]

File: packages/thrift-server-core/src/main/binary.ts (hunk 3 above)
Comment (chad-ramsey-ck) [reply packages/thrift-server-core/src/main/binary.ts:30]: Based on this abstraction it appears so, though there are functions that call out Little Endian explicitly where needed as well: https://github.com/creditkarma/thrift-server/pull/170/files#diff-6b114b0faf6c5f7871f2e3d81b0689e833eccf2c77b204b5db5fe391636cd3adL104

**Pull request merged**